lib/src/git.rs

```diff
@@ -1487,16 +1518,38 @@ impl<'a> GitFetch<'a> {
                         )
                     })
             })
-            .map(|refspec| refspec.map(|r| r.to_git_format()))
-            .collect::<Option<_>>()
-            .ok_or(GitFetchError::InvalidBranchPattern)?;
+            .collect::<Option<Vec<_>>>()
+            .ok_or(GitFetchError::InvalidBranchPattern)
```

Comment:
nit: (at your option) `.ok_or()` can be moved inside `.map()` to remove type annotation from `.collect::<..>()`.